climate_categories/tests/test_conversions.py

```diff
@@ -147,7 +148,7 @@ def test_good_csv(self, tmp_path):
             factors_categories_a={"asdf": 1, "fdsa": 1},
             factors_categories_b={"asdf": 1},
             auxiliary_categories={"aux1": set(), "aux2": set()},
-            csv_line_number=7,
+            csv_line_number=2,
```

Comment:
I think that the line number should be 6 here, no? When we give the user an indication that their file is broken, we do want to tell them the correct line number as shown by emacs (or VS Code, nowadays), not the line number starting when the header is over.

Reply:
Good point :-)

Reply:
Hmm, not sure how to handle this. Since i'm not reading the YAML with a CSV reader anymore, the number is off. Just using the lenghts of YAML keys might be off if someone uses a multiline string.

Reply:
Return a line number from `_read_csv_meta` along the parsed meta data? I can't think of a better way, either.

Reply:
And then of course pass on the line number to `_read_csv_rules` to use as an offset.

Reply:
Hmm... or, wild idea: `fd.seek(0)` before giving it to the `csv.reader` and specify `#` as a comment character. That way, the header will be read and skipped by the `csv.reader`. Don't know if that works.

Reply:
I think CSV reader doesn't handle comments, one has to filter them: https://stackoverflow.com/questions/14158868/python-skip-comment-lines-marked-with-in-csv-dictreader

Reply:
Yeah, that doesn't sound easier than passing around a line number explicitly.

Reply:
I pushed something, it only handles the error message, not when you get it from `conversions.ConversionRuleSpec` in the tests, but i'm not sure if it's necessary for user facing error messages?

Reply:
It is used for example later on in https://github.com/pik-primap/climate_categories/blob/88e4714c1e1aae3a59356b56a185780d641ed7b0/climate_categories/_conversions.py#L503 , which in turn is used to tell users about logical problems in their conversion. So, I think, it should also be corrected in the actual objects.

Just noticed there that a proper integration test which would notice this is missing. Just very focused unit tests, but no test which reads a full conversion from a file and finds an over counting problem, reporting it properly. I should probably add one.

Reply:
I agree, maybe the following works?

https://github.com/pik-primap/climate_categories/pull/50/commits/bb956ffdf59ec92eab0df5c9a0f7399cef212c84

Reply:
Looks like it works, indeed. I have added a test for good error messages and did some stylistic changes, I think it is ready now!